oauth/oauth_config.go

```diff
@@ -20,160 +20,51 @@ import (
 	"fmt"
 	"net/url"
 
-	"github.com/redhat-appstudio/service-provider-integration-operator/api/v1beta1"
-
-	"github.com/redhat-appstudio/service-provider-integration-operator/pkg/logs"
+	"github.com/redhat-appstudio/service-provider-integration-operator/pkg/spi-shared/config"
 	"github.com/redhat-appstudio/service-provider-integration-operator/pkg/spi-shared/oauthstate"
 	"golang.org/x/oauth2"
-	corev1 "k8s.io/api/core/v1"
-	kuberrors "k8s.io/apimachinery/pkg/api/errors"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/log"
-)
-
-const (
-	oauthCfgSecretFieldClientId     = "clientId"
-	oauthCfgSecretFieldClientSecret = "clientSecret"
-	oauthCfgSecretFieldAuthUrl      = "authUrl"
-	oauthCfgSecretFieldTokenUrl     = "tokenUrl"
 )
 
 var (
-	errMissingField            = errors.New("missing mandatory field in oauth configuration")
-	errMultipleMatchingSecrets = errors.New("found multiple matching oauth config secrets")
-	errUnknownServiceProvider  = errors.New("haven't found oauth configuration for service provider")
+	errUnknownServiceProvider = errors.New("haven't found oauth configuration for service provider")
+	errUserConfigNoOAuth      = errors.New("we found user's sp config, but it has no oauth information. This should never happen because we should not have the oauth url at all")
```

Comment:
ok, I see why it is confusing. We're in the oauth service, so we should get here only in cases when user have oauth url => "This should never happen because we should not have the oauth url at all". AND if user has oauth url, we must have oauth configuration somewhere, but we couldn't find it => "we found user's sp config, but it has no oauth information"

Reply:
fixed